File: archive/python/array/BestTimeToSellAndBuyStockII.py
class Solution:
    # @param prices, a list of integer
    # @return an integer
    def maxProfit(self, prices):
        result = 0
        for i in xrange(1, len(prices)):
            diff = prices[i] - prices[i-1]
            if diff > 0:
                result += diff
        
        return result

File: archive/cpp/array/BestTimeToSellAndBuyStockII.cpp
class Solution {
public:
    int maxProfit(vector<int> &prices) {
        int result = 0;
        for (int i = 1; i < int(prices.size()); i++) {
            int diff = prices[i] - prices[i-1];
            if (diff > 0) {
                result += diff;
            }
        }
        return result;
    }
};
